test/k8sT/istio.go

```diff
@@ -0,0 +1,276 @@
+			initPolicyPath := helpers.ManifestGet("istio-sidecar-init-policy.yaml")
+			l7PolicyPath := helpers.ManifestGet("cnp-specs.yaml")
+
+			// Create the L7 policy before creating the pods, in order to test
+			// that the sidecar proxy mode doesn't deadlock on endpoint
+			// creation in this case.
+			// Also create an init policy that allows traffic to Istio services
+			// to ease pod creation.
+			policyPaths = []string{initPolicyPath, l7PolicyPath}
+			for _, policyPath := range policyPaths {
+				By("Creating policy in file %q", policyPath)
+				_, err := kubectl.CiliumPolicyAction(helpers.KubeSystemNamespace, policyPath, helpers.KubectlApply, helpers.HelperTimeout)
+				Expect(err).Should(BeNil(), "Unable to create policy %q", policyPath)
+			}
+
+			resourceYAMLPaths = []string{bookinfoV1YAML, bookinfoV2YAML}
+			for _, resourcePath := range resourceYAMLPaths {
+				By("Creating resources in file %q", resourcePath)
+				res := kubectl.Create(resourcePath)
+				res.ExpectSuccess("Unable to create resource %q: %s", resourcePath, res.CombineOutput())
+			}
+		})
+
+		AfterEach(func() {
+			for _, resourcePath := range resourceYAMLPaths {
+				By("Deleting resource in file %q", resourcePath)
+				// Explicitly do not check result to avoid having assertions in AfterEach.
+				_ = kubectl.Delete(resourcePath)
+			}
+
+			for _, policyPath := range policyPaths {
+				By("Deleting policy in file %q", policyPath)
+				// Explicitly do not check result to avoid having assertions in AfterEach.
+				_ = kubectl.Delete(policyPath)
+			}
+		})
+
+		// shouldConnect checks that srcPod can connect to dstURI.
+		shouldConnect := func(srcPod, dstURI string) bool {
+			By("Checking that %q can connect to %q", srcPod, dstURI)
+			res := kubectl.ExecPodCmd(
+				helpers.DefaultNamespace, srcPod, fmt.Sprintf("%s %s", wgetCommand, dstURI))
+			if !res.WasSuccessful() {
+				GinkgoPrint("Unable to connect from %q to %q: %s", srcPod, dstURI, res.CombineOutput())
+				return false
+			}
+			return true
+		}
+
+		// shouldNotConnect checks that srcPod cannot connect to dstURI.
+		shouldNotConnect := func(srcPod, dstURI string) bool {
+			By("Checking that %q cannot connect to %q", srcPod, dstURI)
+			res := kubectl.ExecPodCmd(
+				helpers.DefaultNamespace, srcPod, fmt.Sprintf("%s %s", wgetCommand, dstURI))
+			if res.WasSuccessful() {
+				GinkgoPrint("Was able to connect from %q to %q, but expected no connection: %s", srcPod, dstURI, res.CombineOutput())
+				return false
+			}
+			return true
+		}
+
+		// formatLabelArgument formats the provided key-value pairs as labels for use in
+		// querying Kubernetes.
+		formatLabelArgument := func(firstKey, firstValue string, nextLabels ...string) string {
+			baseString := fmt.Sprintf("-l %s=%s", firstKey, firstValue)
+			if nextLabels == nil {
+				return baseString
+			} else if len(nextLabels)%2 != 0 {
+				panic("must provide even number of arguments for label key-value pairings")
```

Comment:
please don't panic, set a assert here, the panic is how Ginkgo fails. Use `ginkgo.Fail()` instead.

Reply:
I copied this from Services.go. I will fix this there too.